decko/lib/decko/commands/rake_command.rb

```diff
@@ -16,26 +16,30 @@ def initialize rake_task, args={}
       end
 
       def run
-        commands.each do |cmd|
-          puts cmd
-          # exit_with_child_status cmd
+        puts command
+        # exit_with_child_status cmd
 
-          result = `#{cmd}`
-          process = $?
-          puts result
-          exit process.exitstatus unless process.success?
-        end
+        result = `#{command}`
+        process = $?
+        puts result
+        exit process.exitstatus unless process.success?
+      end
+
+      def command
+        @command ||= command_with_env
       end
 
-      def commands
-        task_cmd = "bundle exec rake #{@task}"
-        return [task_cmd] if !@envs || @envs.empty?
-        @envs.map do |env|
+      def command_with_env
+        @envs.inject(task_cmd) do |task_cmd, env|
```

Comment:
Doesn't this significantly change the logic?

It seems like the old way actually ran the command multiple times. This looks like it makes one command with multiple environments, which doesn't make sense.